device/core/test/_device_client_test.js

```diff
@@ -142,11 +142,163 @@ describe('Device Client', function () {
+    [undefined, null, ''].forEach(function (statusDescription) {
+    it('throws a ReferenceError if \'statusDescription\' is ' + statusDescription + '\'', function(done) {
+      let isSuccess = 0;
+      let statusDescription = 'NaN';
+      let client = new Client(new EventEmitter(), null, {}, {});
+      client._blobStorageUploadParams = { correlationId: 'fakeCorrelationId' };
+      try {
+        client.notifyBlobUploadStatus(isSuccess, statusCode, statusDescription, function() {});
+      } catch (err) {
+        assert.strictEqual(err.name, "ReferenceError");
+        done();
+      }
+      assert.fail('Error not thrown. Error should be thrown.');
+    });
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_41_012: [The `notifyBlobUploadStatus` method shall throw a `ReferenceError` if `correlationId` is not set.]*/
+    it('throws a ReferenceError if \'correlationId\' is not set because getBlobSharedAccessSignature has not been called', function() {
+      let isSuccess = 0;
+      let statusCode = 200;
+      let statusDescription = 'NaN';
+      var client = new Client(new EventEmitter(), null, {}, {});
+      assert.throws(function() {
+        client.notifyBlobUploadStatus(isSuccess, statusCode, statusDescription, function() {});
+      });
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_41_013: [The `notifyBlobUploadStatus` method shall call the `done` callback with an `Error` object if the notify fails.]*/
+    it('calls the done callback with an Error object if the notify fails', function(done) {
+      let isSuccess = false;
+      let statusCode = 0;
+      let statusDescription = 'NaN';
+      let FakeFileUploadApi = function (uploadResult) {
+        this.notifyUploadComplete = function(correlationId, uploadResult, callback) {
+          callback(new Error('fake notify error'));
+        };
+      };
+      let client = new Client(new EventEmitter(), null, null, new FakeFileUploadApi());
+      client._blobStorageUploadParams = { correlationId: 'fakeCorrelationId' };
+      client.notifyBlobUploadStatus(isSuccess, statusCode, statusDescription, function(err) { 
+      assert.instanceOf(err, Error);
+        done();
+      });
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_41_014: [The `notifyBlobUploadStatus` method shall call the `done` callback with no parameters if the notify succeeds.]*/
+    it('calls the done callback with no parameters if the upload succeeded', function (done) {
+      let isSuccess = true;
+      let statusCode = 200;
+      let statusDescription = 'NaN';
+      let FakeFileUploadApi = function (uploadResult) {
+        this.notifyUploadComplete = function(correlationId, uploadResult, callback) {
+          callback();
+        };
+      };
+      let client = new Client(new EventEmitter(), null, null, new FakeFileUploadApi());
+      client._blobStorageUploadParams = { correlationId: 'fakeCorrelationId' };
+      client.notifyBlobUploadStatus(isSuccess, statusCode, statusDescription, done);
+    });
+  });
```

Comment:
Please add two other tests: 
1. `it('calls getBlobSharedAccessSignature 10 times to upload 10 parallel files and succeeds notifying all 10 uploads in the same order')`
1. `it('calls getBlobSharedAccessSignature 10 times to upload 10 parallel files and succeeds notifying all 10 uploads in reverse order')`